winc-rs/src/client/tcp_stack.rs

```
@@ -188,9 +188,7 @@ impl<X: Xfer> embedded_nal::TcpClientStack for WincClient<'_, X> {
                         }
                         SocketError::Timeout => {
                             debug!("Timeout on receive");
-                            // Timeouts just get turned into a further wait
-                            // Todo: Maybe re-send the command ?
-                            Err(StackError::ContinueOperation)
```

I verified - yes this breaks `tcp_shell` - but also, tcp_shell wasn't working correctly before this change either.

The `recv` command needs to be acually re-sent here for the infinite wait to work.

---

I have a fix here #67

---

I saw the fix, and I would argue that we should return an error instead of resending the command. This way, the user will know that the WiFi module is in the correct state. If we resend the command and the user (who may be tracking time in the application) decides to send another command, the module behavior may become unpredictable.

What we can do is set the default infinite timeout to 0, which aligns with the `embedded-nal` trait, and provide an option for the user to exit the function as you suggested in [issue 66](https://github.com/kaidokert/winc-rs/issues/66)